lib/active_record/connection_adapters/mysql_flexmaster_adapter.rb

```diff
@@ -29,8 +29,9 @@ class TooManyMastersException < StandardError; end
       class NoServerAvailableException < StandardError; end
 
       CHECK_EVERY_N_SELECTS = 10
-      DEFAULT_CONNECT_TIMEOUT = 5
-      DEFAULT_TX_HOLD_TIMEOUT = 5
+      DEFAULT_CONNECT_TIMEOUT = 2
+      DEFAULT_CONNECT_RETRIES = 2
+      DEFAULT_TX_HOLD_TIMEOUT = 6
```

Comment:
This was related for me going for a higher value on DEFAULT_CONNECT_TIMEOUT earlier. I've lowered the connect timeout and reset the tx_hold.